src/components/Map/Map.js

```diff
@@ -3,39 +3,133 @@ import "leaflet-defaulticon-compatibility";
 import "leaflet-defaulticon-compatibility/dist/leaflet-defaulticon-compatibility.css";
 import "leaflet/dist/leaflet.css";
 import { useMapEvents } from "react-leaflet";
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import CustomPopup from "../Popup/Popup";
 import Tools from "../Tools/Tools";
+import * as L from "leaflet";
 
 const initialMarkers = [];
 
+const LeafIcon = L.Icon.extend({
+  options: {},
+});
+const blueIcon = new LeafIcon({
+    iconUrl:
+      "https://chart.apis.google.com/chart?chst=d_map_pin_letter&chld=%E2%80%A2|FFA500&chf=a,s,ee00FFFF",
+  }),
+  greenIcon = new LeafIcon({
+    iconUrl:
+      "https://chart.apis.google.com/chart?chst=d_map_pin_letter&chld=%E2%80%A2|2ecc71&chf=a,s,ee00FFFF",
+  });
+
+function getIcon(marker) {
+  if (marker.isVisited) {
+    return greenIcon;
+  }
+  return blueIcon;
+}
+
 function LocationMarker({ coords }) {
   const map = useMap();
   map.flyTo([coords.geometry.lat, coords.geometry.lng], 13);
   return null;
 }
 
+async function createMarker(marker) {
+  const response = await fetch("/api/places", {
+    method: "POST",
+    body: JSON.stringify(marker),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  const data = await response.json();
+  return data;
+}
+
+async function updateMarker(id, marker) {
+  const response = await fetch(`/api/places/${id}`, {
+    method: "PUT",
+    body: JSON.stringify(marker),
+    headers: {
+      "Content-Type": "application/json",
+    },
+  });
+  const data = await response.json();
+  return data;
+}
+
 function Map({ searchResult, isAddingMarker }) {
   const map = useMapEvents({
     click: (e) => {
       if (!isAddingMarker) {
         return;
       }
-      console.log("clicked", e);
+
       handleAddMarker({
-        id: crypto.randomUUID(),
+        //   _id: crypto.randomUUID(),
         lat: e.latlng.lat,
-        long: e.latlng.lng,
+        lng: e.latlng.lng,
+        isVisited: false,
+        stars: 0,
+        image: "",
+        expirience: "",
       });
     },
   });
   const [markers, setMarkers] = useState(initialMarkers);
-  function handleAddMarker(marker) {
+
+  useEffect(() => {
+    async function getPlaces() {
+      const response = await fetch("/api/places");
+      const data = await response.json();
+      setMarkers(data);
+    }
+    getPlaces();
+  }, []);
+
+  async function handleAddMarker(data) {
+    const marker = await createMarker(data);
     setMarkers([...markers, marker]);
   }
 
   function handleDeleteMarker(markerId) {
-    setMarkers(markers.filter((marker) => marker.id !== markerId));
+    setMarkers(markers.filter((marker) => marker._id !== markerId));
```

Comment:
I see that your update function below actually makes a request to an api route that then updates the database. Why doesn't the delete function do something similar?

Reply:
I am still missing the request to the api route that will then delete the item from the database. At the moment, as far as i can see, we only delete the marker from the local state...